Topics/User_Experience/Design_Requirements.md

```diff
@@ -0,0 +1,120 @@
+
+### **3. Communication**
+
+Accurately defining and documenting requirements makes sure that the communication between team members and clients are not lost and can be referred to later in the project to see if they have been met, especially after usability studies. Benefits include;
+
+- **Clearer communication**: Defining requirements ensures that the developer team and the client are on the same page and they can identify misunderstandings early on.
+- **Asynchronous communication**: With shared understanding and written documentation, certain questions can be addressed asynchronously, to see if a certain feature matches the clients design requirements.
+- **Efficiency**: There is less time spent on actively thinking about what the client had wanted. The written documentation can be easily referred to in later processes and can be used to hold clients accountable as well.
+
+## **Functional Design Requirements**
+
+Functional design requirements specify the goals of the desired product. They focus on the “usefulness” of the product.
+
+Examples include;
+- The system must enable users to verify their accounts using their phone number.
+- The system should support backward and forward navigation in the research article to follow the natural order of sections.
+- The system must allow visitors to sign up for the newsletter by inputting their email. 
+
+### **Relation to User Stories**
+
+Functional requirements are not the same as user stories. In fact, user stories can be used to derive design requirements while centering user needs. An [example](https://www.nuclino.com/articles/functional-requirements) of this is as follows: 
+
+“User story: As an existing user, I want to be able to log into my account.
+
+Functional requirements:
+- The system must allow users to log into their account by entering their email and password.
+- The system must allow users to log in with their Google accounts.
+- The system must allow users to reset their password by clicking on "I forgot my password" and receiving a link to their verified email address.” [2]
+
+## **Non-Functional Design Requirements**
+
+Non-functional design requirements specify the methods used to achieve the goals of the desired product. They focus on the “usability” of the product and provide constraints on the system and its development. 
+
+Examples include;
+- The system should ensure security and privacy in the account verification process, adhering to industry standards for data protection and encryption.
+- The system should display as much text as possible, without compromising accessibility standards.
+- When the user presses the sign-up button, the system should load the confirmation screen in two seconds.
+
+As seen in the examples, non-functional design requirements impose constraints on how the functional requirements are implemented in terms of *performance, security, reliability, scalability, and portability.*
+
+## **Extracting Action Items**
```

Comment:
This very detailed example of extracting action items is very useful in understanding the concept at hand. I think CSC301 students will strongly benefit from this section.